plugins/amqp/src/plugin.cpp

```diff
@@ -600,8 +623,22 @@ class rabbitmq_saver final : public plugin_saver {
       .mandatory = args_.mandatory,
       .immediate = args_.immediate,
     };
-    return [&ctrl, engine = std::make_shared<amqp_engine>(std::move(*engine)),
-            opts = std::move(opts)](chunk_ptr chunk) mutable {
+    auto hb = try_get<uint64_t>(config_, "heartbeat");
+    if (hb && *hb && **hb > 0) {
+      // If we are requesting heartbeats, we are also responsible to handle the
+      // heartbeats we get. If we have long gaps in interaction with the broker,
+      // we need to proactively check explicitly if there is something for us.
+      // We check 3 times per heartbeat interval, at most once per second.
+      auto interval = std::max((int64_t) 1,
+        detail::narrow_cast<int64_t>(std::floor(**hb/3)));
+      TENZIR_DEBUG("using heartbeat interval {}", interval);
+      detail::weak_run_delayed_loop(&ctrl.self(),
+        std::chrono::seconds{interval}, [engine] {
+        TENZIR_DEBUG("processing heartbeats");
```

Comment:
I'm not sure if this is the right log level or even the right place. Wouldn't this amount to a spam of `processing heartbeats`  at the `TENZIR_DEBUG` level?

Maybe this can be `TENZIR_TRACE`?

Reply:
Yes. This is also the case for https://github.com/tenzir/tenzir/blob/main/plugins/amqp/src/plugin.cpp#L324 in which `consuming message` is also spammed a lot due to the (intended!) run/timeout loop. So I assumed it's fine to do that ;)

Reply:
I'd be happy to make both `TENZIR_TRACE` TBH.

Reply:
Good idea. I'd say we should go for it.